includes/class-wc-payments-explicit-price-formatter.php

```diff
@@ -23,6 +33,66 @@ public static function init() {
 		add_action( 'woocommerce_admin_order_totals_after_total', [ __CLASS__, 'unregister_formatted_woocommerce_price_filter' ] );
 	}
 
+	/**
+	 * Overrides the MultiCurrency instance that the class uses.
+	 * Mostly for testing purposes.
+	 *
+	 * @param   MultiCurrency $multi_currency  MultCurrency class.
+	 *
+	 * @return  void
+	 */
+	public static function set_multi_currency_instance( MultiCurrency $multi_currency ) {
+		self::$multi_currency_instance = $multi_currency;
+	}
+
+	/**
+	 * Checks if the method should output explicit price on frontend
+	 *
+	 * @return  bool  Whether if it should return explicit price or not
+	 */
+	private static function should_output_explicit_price() {
+		// As is_admin() returns false for REST requests, we need to skip those checks for REST requests for backend too.
+		$is_backend_request = (
+			// Current URL is an admın URL.
+			( 0 === strpos( strtolower( wp_get_referer() ), strtolower( admin_url() ) ) )
+			// The current request is a REST request.
+			&& WC()->is_rest_api_request()
+		);
+
+		// Only apply this for frontend.
+		if ( ! is_admin() && ! defined( 'DOING_CRON' ) && ! $is_backend_request ) {
+			// If customer multi currency is disabled, don't use explicit currencies on frontend.
+			// Because it'll have only the store currency active, same as count == 1.
+			if ( ! WC_Payments_Features::is_customer_multi_currency_enabled() ) {
+				return false;
+			}
+
+			// If the MultiCurrency instance hasn't been defined yet, fetch the instance.
+			if ( null === self::$multi_currency_instance ) {
+				self::$multi_currency_instance = WC_Payments_Multi_Currency();
+			}
+
+			// If the instance isn't initalized yet, skip the checks.
+			if ( false === self::$multi_currency_instance->is_initialized() ) {
+				return false;
+			}
+
+			$enabled_currencies = self::$multi_currency_instance->get_enabled_currencies();
+
+			// If there isn't any enabled currency, skip it.
+			if ( empty( $enabled_currencies ) ) {
+				return false;
+			}
+
+			// Don't attach explicit price filters on frontend with a single currency setup.
+			if ( is_array( $enabled_currencies ) && 1 === count( $enabled_currencies ) ) {
```

Comment:
Should we do `1 <= count( $enabled_currencies )` here instead, just in case an empty array is returned for some reason?

Reply:
The condition above it checks if it's an empty array?

Reply:
So it does. 😅